core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -462,10 +464,12 @@ class ContainerPool(
                 // We grab the size first so we know there has been enough delay for anything we are shutting down
                 val size = toBeRemoved.size()
                 1 to size foreach { _ =>
-                    val ci = toBeRemoved.poll()
-                    if (ci != null) { // should never happen but defensive
+                    val removeJob = toBeRemoved.poll()
+                    if (removeJob != null) {
```

Comment:
Scala-style comment ahead. Ignore if you like the null-check better.

Would it make sense to wrap the `poll` in an `Option` instead of `null`-checking?

```
Option(toBeRemoved.poll()) match {
    case Some(removeJob) =>
        Thread.sleep(100)
        teardownContainer(removeJob)
    case None => error(this, "toBeRemove.poll failed - possibly another concurrent remover?")
}
```